From 7d603857de45d778ef1877645f3e9adbb21fe93d Mon Sep 17 00:00:00 2001
From: prasannas <prasannas@ami.com>
Date: Mon, 16 Oct 2023 08:34:20 +0530
Subject: [PATCH 54/54] Fixed the unable to set user lockout time manual

Test:
1)Set User lockout time as Manual 
Patch call
https://x.x.x.x/redfish/v1/AccountService

Body:
{
    "AccountLockoutDuration": 0,
    "AccountLockoutThreshold": 10
}

Response:
{
    "@Message.ExtendedInfo": [
        {
            "@odata.type": "#Message.v1_1_1.Message",
            "Message": "The request completed successfully.",
            "MessageArgs": [],
            "MessageId": "Base.1.16.0.Success",
            "MessageSeverity": "OK",
            "Resolution": "None"
        }
}
2)Updating Username
Post call
https://x.x.x.x/redfish/v1/AccountService/Accounts/test1
Body:
{
    "UserName":"test2",
}
Response:
{
    "@Message.ExtendedInfo": [
        {
            "@odata.type": "#Message.v1_1_1.Message",
            "Message": "The resource has been created successfully.",
            "MessageArgs": [],
            "MessageId": "Base.1.16.0.Created",
            "MessageSeverity": "OK",
            "Resolution": "None."
        }
    ]
}
---
 redfish-core/lib/account_service.hpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/redfish-core/lib/account_service.hpp b/redfish-core/lib/account_service.hpp
index 8c9f2c5..ab16886 100644
--- a/redfish-core/lib/account_service.hpp
+++ b/redfish-core/lib/account_service.hpp
@@ -1695,13 +1695,13 @@ inline void handleAccountServicePatch(
     {
         // Account will be locked permanently after the N number of failed login
         // attempts if we set unlockTimeout value to be 0.
-        if (unlockTimeout.value() == 0)
+        /*if (unlockTimeout.value() == 0)
         {
             BMCWEB_LOG_INFO("Unlock timeout value must be greater than zero");
             messages::propertyValueNotInList(asyncResp->res, "unlockTimeout",
                                              "AccountLockoutDuration");
             return;
-        }
+        }*/
 
         sdbusplus::asio::setProperty(
             *crow::connections::systemBus, "xyz.openbmc_project.User.Manager",
@@ -2325,6 +2325,7 @@ inline void
 
         updateUserProperties(asyncResp, newUser, password, enabled, roleId,
                              locked, accountTypes, userSelf);
+        messages::success(asyncResp->res);
     },
         "xyz.openbmc_project.User.Manager", "/xyz/openbmc_project/user",
         "xyz.openbmc_project.User.Manager", "RenameUser", username,
-- 
2.9.5

